staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -169,7 +213,7 @@ func (q *delayingType) AddAfter(item interface{}, duration time.Duration) {
 
 	// immediately add things with no delay
 	if duration <= 0 {
-		q.Add(item)
+		q.immediateAddPreemptsWaiting(item)
```

Comment:
Given that this function needs to lock, is there really a good reason to have this special case any more? What if we stuff it in the channel no matter what?

Reply:
It only needs to lock if someone calls AddAfter() with a 0 second wait duration - which I have taken to mean "Add this immediately".  I am striving to make all of the immediate add's behave the same and so am using the `immediateAddPreemptsWaiting` for the immediate add everywhere, including the waitingLoop when performing an immediate, with the exception of `AddIfNotWaiting()` which is now a special case.

Reply:
I think for me personally 0 is very close to 1ns, so it's a bit weird to handle them so differently. I'd consider just removing the special case completely, maybe move it to the processing queue and change the threshold to be, I don't know, 1us?